README.md

```diff
@@ -1,277 +1,899 @@
+One-way swaps use one universal validator script for all swap pairs, and one universal minting
+policy for all one-way beacons. There are two beacons for one-way swaps: the trading pair beacon and
+the offer beacon.
 
-### The Cardano-Swaps Protocol
-Cardano-Swaps takes inspiration from Axo's programmable swaps design, adds delegation control as a foundational feature, and, through the use of Beacon Tokens, removes the need for specialized indexers. The only remaining bottleneck is the querying capacity of existing off-chain APIs, such as Blockfrost or Koios. (This is not a limitation for users with powerful enough hardware, as they can run their own API database).
+The offer beacon asset name is `sha2_256( offer_policy_id ++ offer_asset_name )`.
 
-The Cardano-Swaps Protocol is broadly comprised of two steps:
+The trading pair beacon asset name is 
+`sha2_256( offer_id ++ offer_name ++ ask_id ++ ask_name )`. However, in the case where one
+of the assets is ADA (which is just the empty bytestring), the ADA policy id is set to "00" instead.
+Without this changed, there would be no difference between the offer beacon and trading pair beacon
+when ADA is part of a swap.
 
-1. **Prepare Swap Address** - Alice prepares a swap address for swapping Token "ABC" for Token "XYZ" by minting an ABC:XYZ Beacon Token in a "Beacon UTxO" that designates her address specifically as an ABC-for-XYZ swap address. She then (in the same or separate transaction) outputs one or more "Swap UTxO(s)" with "ABC" Tokens, with an attached `Price` Datum that designates her desired price (in "XYZ") for that particular UTxO.
-   
-2. **Execute Swap(s)** - Bob queries a list of ABC-for-XYZ swap addresses (via the ABC:XYZ Beacon Token), and finds some attractively priced UTxOs in Alice's swap address. He submits a transaction that consumes one (or multiple) of Alice's "Swap UTxO(s)", and one or more of his own UTxO(s) containing Alice's requested "XYZ" Token. Bob can then output Alice's "ABC" Tokens to himself in proportion to the amount of "XYZ" tokens he outputs back to Alice. Bob does not have to buy all of the contents of one of Alice's UTxOs - he buy however much he wants, and output the remaining "ABC" back to Alice in the same UTxO as her "XYZ" payment. This UTxO (containing Alice's "XYZ" tokens) remains swappable as long as some "ABC" tokens remain in it; nobody except Alice can claim the "XYZ" tokens.
+These beacons allow for expressive queries since it makes it possible to query all swaps where XYZ
+is being offered, regardless of the asset being asked for. This is in addition to being able to
+query all swaps for a specific trading pair. This extra expressiveness improves the arbitrager's
+ability to find profitable paths through currently open swaps.
```

Comment:
When searching there is only 1 offer_policy_id and you know the offer_asset_name you want so you calculate the sha2_256 sum and search for this hash value which represents the offer beacon?  Are you using the sha2_256 sum to limit the beacon name size rather than simply using the offer_policy_id + offer_asset_name directly?

Reply:
> Are you using the sha2_256 sum to limit the beacon name size rather than simply using the offer_policy_id + offer_asset_name directly?

That is correct. Token names are capped at 64 characters and policy ids take up 56 of the 64 characters so simply concatenating them is not possible.